### PR TITLE
Shared Block Device Support

### PR DESCRIPTION
**Pull request by mhenriks**
Add --shared-block-device param to gocli that will:

1. Create shared docker volume
2. node01 will create disk images on shared volume when node starts
3. Devices made available on each node

First step for what KubeSAN support

https://gitlab.com/kubesan/kubesan

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
gocli support for shared block devices
```

### CODE REVIEW

File: cluster-provision/gocli/cmd/run.go [hunk 1]
```diff
@@ -652,6 +665,20 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
...
 
+		var vmArgsSharedDisks []string
+		if len(sharedDisks) > 0 {
+			for i, size := range sharedDisks {
+				pciOffset := i + 1
+				resource.MustParse(size)
+				disk := fmt.Sprintf("/shared/disk%d.img", i)
+				blockDev := fmt.Sprintf("-blockdev file,filename=%s,node-name=shared-disk-%d,read-only=off,cache.direct=on,cache.no-flush=off", disk, i)
+				device1 := fmt.Sprintf("-device pcie-root-port,id=pci.%d,bus=pcie.0", pciOffset)
+				device2 := fmt.Sprintf("-device virtio-blk-pci,bus=pci.%d,drive=shared-disk-%d,id=shared-disk-%d,share-rw=on,write-cache=on,werror=stop,rerror=stop", pciOffset, i, i)
```
Comment (mhenriks) [inline cluster-provision/gocli/cmd/run.go:676]: @alicefr can you validate these settings?

File: cluster-provision/gocli/cmd/run.go [hunk 2]
```diff
@@ -557,6 +561,15 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
 		}
 	}
 
+	sharedVolumeName := prefix + "-shared"
+	if len(sharedDisks) > 0 {
+		sharedVolume, err := cli.VolumeCreate(ctx, volume.CreateOptions{Name: sharedVolumeName})
```
Comment (mhenriks) [inline cluster-provision/gocli/cmd/run.go:566]: rm.go (cluster-down) will clean up this volume. Technically any volume with the prefix

File: cluster-provision/gocli/cmd/run.go (hunk 1 above)
Comment (alicefr) [reply cluster-provision/gocli/cmd/run.go:676]: @mhenriks I would disable the cache, `write-cache=off`

Comment (mhenriks) [reply cluster-provision/gocli/cmd/run.go:676]: Hi @alicefr yeah I was a little confused about that setting. I actually took the settings from a kubevirt `shareable` disk. Was surprised to see `write-cache=on`. But then I checked qemu man page [1] and I got the impressions, which may be wrong. Correct me if so!

1. `write-cache` is synonymous with `cache.writeback` on device
2. `none` cache is `cache.writeback=on`, `cache.direct=on`, `cache.no-flush=off`

[1] https://www.qemu.org/docs/master/system/qemu-manpage.html

Comment (alicefr) [reply cluster-provision/gocli/cmd/run.go:676]: sorry, for the noise, I think you are right, at least from the QEMU doc. Thehn, just ignore my comment

**Review by awels — APPROVED** [review]

**Review by brianmcarey — APPROVED** [review]
/approve

**Pull request merged**